src/if_statements.py

```diff
@@ -600,27 +648,56 @@ def emit_return(context: Context, node: ReturnNode, body: Body) -> None:
 
 def build_switch_between(
     context: Context,
-    start: SwitchNode,
+    switch: SwitchNode,
     end: Node,
```

Comment:
Pre-existing and more of a question than a review comment, but I figure I should ask anyway: when the switch contains an early return, will that result in `end` being a `TerminalNode`? Won't that throw a lot of control flow construction off?

Regardless of answer I think cases with early returns would probably make for good test cases!

Reply:
Yeah, `end` would be the `TerminalNode`. The cases will have to end in `goto` instead of `break;`

This also sort of happens with `ConditionalNode`s as well, but when the if body ends in a jump, the else is elided.